pkg/webhook/webhook.go

```diff
@@ -832,3 +835,44 @@ func SetInjectHugepageDownApi(hugepageFlag bool) {
 func SetHonorExistingResources(resourcesHonorFlag bool) {
 	honorExistingResources = resourcesHonorFlag
 }
+
+//Watch blocks until either TLS cert & key watcher or HTTP server or termination signal generated
+func Watch(server nri.Service, watcher nri.Service, term chan os.Signal) (err error) {
+	signal.Notify(term, syscall.SIGHUP, syscall.SIGINT, syscall.SIGTERM, syscall.SIGQUIT)
+	select {
+	case <-watcher.StatusSignal(): // when TLS cert & key watcher finishes unexpectedly
+		glog.Error("TLS key & cert watcher ended")
+		err = server.Quit()
+	case <-server.StatusSignal(): // when HTTP server finishes unexpectedly
```

Comment:
a question: when write happens on this server channel ? i couldn't find it anywhere  in the code :)

Reply:
@pperiyasamy see server.go: within method with pointer receiver ```Run()``` -> ```defer mSrv.Status.Close()```. When the channel closes: 

> After calling close, and after any previously sent values have been received, receive operations will return the zero value for the channel's type without blocking.
 

ref: https://golang.org/ref/spec#Close

.. this causes this case to trigger

Reply:
ah! ok, that means server.go's` mSrv.instance.Start() `is blocking call, right ?